### ISSUE TITLE
"cannot satisfy -package hspec-2.7.1" (test dependencies aren't built)

### ISSUE THREAD

**Issue by runeksvendsen** [dramforever/vscode-ghc-simple#51]
## Status

- [x] A workaround is available
- [ ] A fix is written
- [ ] A fix is released
- [ ] The fixed version is on the marketplace.

---

- Version of vscode-ghc-simple: 0.1.13
- Version of VSCode: 1.42.1 (commit: c47d83b293181d9be64f27ff093689e8e7aed054)

- Version of Haskell-related tools:
    
    - GHC: 8.6.5
    - Stack: 2.1.3 (commit: 0fa51b9925decd937e4a993ad90cb686f88fa282)
    - Cabal: 2.4.1.0@rev:2 (lts-14.0)

## Workaround

Manually run `stack test --no-run-tests` inside the project directory before opening VS Code.

## Steps to reproduce

1. Start VS Code for the following project (with an empty build cache): https://github.com/runeksvendsen/crypto-orderbook-db/commit/017322d822bb288531c043783a34708847e83e63

2. VS Code / Output / GHC:

```
...
ghci | <command line>: cannot satisfy -package hspec-2.7.1
ghci |     (use -v for more information)
```

I expect no "cannot satisfy -package" error.

## Workspace

1. Repo: https://github.com/runeksvendsen/crypto-orderbook-db/commit/017322d822bb288531c043783a34708847e83e63
2. Stack version: 2.1.3 (Git revision 0fa51b9925decd937e4a993ad90cb686f88fa282)
3. Start VS Code (with `vscode-ghc-simple` enabled)

## GHC log

https://gist.github.com/runeksvendsen/c62d0356afa98202953a6c197eda3927

**Comment by runeksvendsen** [comment]
So, it seems like the command that vscode-ghc-simple runs  for my project (`stack --color never repl --no-load crypto-orderbook-db:lib crypto-orderbook-db:exe:crypto-orderbook-db crypto-orderbook-db:exe:crypto-orderbook-db-create-tables crypto-orderbook-db:test:crypto-orderbook-db-test`) does not trigger building of the test dependencies.

Running the above command results in the aforementioned error:

```
Building all executables for `crypto-orderbook-db' once. After a successful build of all of them, only specified executables will be rebuilt.
crypto-orderbook-db> initial-build-steps (lib + exe)
The following GHC options are incompatible with GHCi and have not been passed to it: -threaded
Configuring GHCi with the following packages: crypto-orderbook-db
GHCi, version 8.6.5: http://www.haskell.org/ghc/  :? for help
<command line>: cannot satisfy -package hspec-2.7.1
    (use -v for more information)
```

whereas if the "test" project is removed from the command (i.e. `stack --color never repl --no-load crypto-orderbook-db:lib crypto-orderbook-db:exe:crypto-orderbook-db crypto-orderbook-db:exe:crypto-orderbook-db-create-tables`) it works without problems.

If there is a test target included as one of the targets it appears we need to pass the `--test` argument to the `stack` command, in order to trigger a build of test dependencies (and thus avoid the error).

**Comment by dramforever** [comment]
> If there is a test target included as one of the targets it appears we need to pass the --test argument to the stack command, in order to trigger a build of test dependencies

This seems to be the case. I'll need to investigate more about stack's behavior and specifically how it handles targets.

**Comment by dramforever** [comment]
Sorry for the extremely delayed response. I've recently started working on supporting multi-component/multi-package projects. I think your issue might be helped by this.

If you haven't abandoned vscode-ghc-simple, please consider testing out development versions of the `vgs-bios` branch. See more information here: <https://github.com/dramforever/vscode-ghc-simple/discussions/87>.

In short, you will need to install a development version of the `vgs-bios` branch and set up an `hie.yaml` file, with the same format as the one accepted by [hie-bios] to tell vscode-ghc-simple about your project structure. If you need help or find any problems please do consider reaching out.

[hie-bios]: https://github.com/mpickering/hie-bios#explicit-configuration

Not everything is well-supported yet, but hopefully simple multi-component or multi-package setups will work. The same file should also help out those who use ghcide. Many projects already have an `hie.yaml` file, so these will hopefully work out of the box. See the discussion page linked above for details for more information.

Thanks for considering helping out!

**Comment by dramforever** [comment]
I'm currently still working on seeing if test targets work correctly. I see that the project you gave already has an `hie.yaml` file, so I'll be trying it out.

**Comment by dramforever** [comment]
Released in v0.2.0
